ARK Server Manager/Lib/ServerRCON.cs

```diff
@@ -370,13 +370,45 @@ private void HandleCommand(ConsoleCommand command)
                     }
                 }).DoNotWait();
             }
+            else
+            {
+                var arkData = await ArkData.ArkDataContainer.CreateRemoteAsync();
+                arkData.SteamLoaded = true;
+                //await arkData.LoadSteamAsync(Config.Default.SteamAPIKey);
+                TaskUtils.RunOnUIThreadAsync(() =>
+                {
+                    foreach (var playerData in arkData.Players)
+                    {
+                        var playerToUpdate = this.Players.FirstOrDefault(p => p.SteamId == Int64.Parse(playerData.SteamId));
+                        if (playerToUpdate != null)
+                        {
+
+                            playerToUpdate.AvatarImage = new BitmapImage(new Uri(playerData.AvatarUrl));
```

Comment:
If this performs a network operation, you do not want to do this on the UI thread.  All network data transfers should be done on a background thread, and when they are complete, then you can load the image from that data.